src/iterator.jl

```diff
@@ -0,0 +1,38 @@
+struct CuIterator{B}
+    batches::B
+    initial_pool_size::Ref{Int}
+    pool::Ref{Mem.DeviceBuffer}
```

Comment:
Why an immutable struct with `Ref` and not just a mutable struct?

Reply:
Didn't want anybody mutating `batches` on me, but that's probably overly defensive on my part; will change it